images/operator/Dockerfile

```diff
@@ -69,3 +71,35 @@ COPY --from=builder /out/${TARGETOS}/${TARGETARCH}/LICENSE.all /LICENSE.all
 WORKDIR /
 ENV GOPS_CONFIG_DIR=/
 CMD ["/usr/bin/cilium-${OPERATOR_VARIANT}"]
+
+# Intermediate stage because expansion is not working in COPY --from
+FROM --platform=${BUILDPLATFORM} ${CILIUM_BUILDER_IMAGE} as debug-builder
```

Comment:
`COPY --from=${CILIUM_BUILDER_IMAGE} /go/bin/dlv /usr/bin/dlv`

results in 

```
Dockerfile:98
--------------------
  96 |     CMD ["/usr/bin/cilium-${OPERATOR_VARIANT}"]
  97 |     
  98 | >>> COPY --from=${CILIUM_BUILDER_IMAGE} /go/bin/dlv /usr/bin/dlv
  99 |     RUN mv /usr/bin/cilium-${OPERATOR_VARIANT} /usr/bin/cilium-${OPERATOR_VARIANT}-bin
 100 |     ENV DEBUG_PORT=2346
--------------------
ERROR: failed to solve: failed to parse stage name "${CILIUM_BUILDER_IMAGE}": invalid reference format: repository name must be lowercase
```

even though the following would work (but would break the cilium runtime/builder image update logic.

`COPY --from=quay.io/cilium/cilium-builder:f55691636922518ca40802f59d57f36ee4741c0d@sha256:4c21a3e21e9e07f0301191b3ae6921c8208a3910ce1b73be6ad445e544590e3a /go/bin/dlv /usr/bin/dlv`

the builder stage of `cilium/Dockerfile` already builds on top of the cilium builder image - there doesn't have the exact same need.

but the dockerfile of cilium builder itself also has to redefine all images as stages to be able to copy files in other stages. (https://github.com/cilium/cilium/blob/master/images/builder/Dockerfile#L4-L16)

Reply:
In the cilium Dockerfile, we use `COPY --from=builder ...` rather than referencing a variable. Could we just do that?

Reply:
(overall I'm not too worried, I'm just thinking that it'd be good to minimize the differences between the release build vs. debug builds, and reduce duplicated statements. If we ever change things, it's easy to accidentally make the release vs. debug build diverge)

Reply:
> In the cilium Dockerfile, we use COPY --from=builder ... rather than referencing a variable. Could we just do that?

well the `builder` is just a build stage itself - which uses a variable when being defined too. so actually, no difference between the operator and agent.

> good to minimize the differences between the release build vs. debug builds, and reduce duplicated statements

strongly agree on that one. going to rename the stage from `debug-builder` to `debug-tools` where we then copy dlv & debug-wrapper from - only in case of the debugger image of course. should be fine then.